File: app/src/main/java/com/codemybrainsout/imageviewer/api/UserService.java
package com.codemybrainsout.imageviewer.api;

import com.codemybrainsout.imageviewer.model.Collection;
import com.codemybrainsout.imageviewer.model.Photo;
import com.codemybrainsout.imageviewer.model.User;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by ahulr on 07-06-2017.
 */

public interface UserService {

    @GET("/users/{username}")
    Observable<User> getUser(@Path("username") String user);

    @GET("/users/{username}/photos")
    Observable<List<Photo>> getUserPhotos(@Path("username") String user, @Query("page") int page, @Query("per_page") int limit);

    @GET("/users/{username}/likes")
    Observable<List<Photo>> getUserLikes(@Path("username") String user, @Query("page") int page, @Query("per_page") int limit);

    @GET("/users/{username}/collections")
    Observable<List<Collection>> getUserCollections(@Path("username") String user, @Query("page") int page, @Query("per_page") int limit);

}

File: app/src/main/java/com/codemybrainsout/imageviewer/api/UserService.kt
package com.codemybrainsout.imageviewer.api

import com.codemybrainsout.imageviewer.model.Collection
import com.codemybrainsout.imageviewer.model.Photo
import com.codemybrainsout.imageviewer.model.User
import io.reactivex.rxjava3.core.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Created by ahulr on 07-06-2017.
 */
interface UserService {
    @GET("/users/{username}")
    fun getUser(@Path("username") user: String?): Observable<User>

    @GET("/users/{username}/photos")
    fun getUserPhotos(
        @Path("username") user: String?,
        @Query("page") page: Int,
        @Query("per_page") limit: Int
    ): Observable<List<Photo>>

    @GET("/users/{username}/likes")
    fun getUserLikes(
        @Path("username") user: String?,
        @Query("page") page: Int,
        @Query("per_page") limit: Int
    ): Observable<List<Photo>>

    @GET("/users/{username}/collections")
    fun getUserCollections(
        @Path("username") user: String?,
        @Query("page") page: Int,
        @Query("per_page") limit: Int
    ): Observable<List<Collection>>
}